### PR TITLE
feat: added .new (dot-new) extension

### PR DESCRIPTION
**Pull request by natterstefan**
## Description

I am a fan of `.new` Domains ([what's that?](https://whats.new/)). That's why I added an extension for that to quickly start new things!

## Screencast

dot-new-01

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `metadata` folder

### CODE REVIEW

File: extensions/dot-new/scripts/generate-dataset.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  const data = flattenDeep(
+    tablesAsJson.map((tables) =>
+      tables.map((table) => {
+        const { 0: urls, 1: provider, 2: text } = table;
+        const prepUrls = urls.split(" ").filter(Boolean);
+
+        return prepUrls.map((url) => ({ url, provider, text }));
+      })
+    )
+  );
```
Comment (natterstefan) [inline extensions/dot-new/scripts/generate-dataset.ts:22]: I get the list of .new domains (and their data) directly from the tables of shortcuts on the [official website](https://whats.new/shortcuts/).

File: extensions/dot-new/src/index.tsx [hunk 2]
```diff
@@ -12,6 +12,12 @@ export function getFilteredDataset(query?: string): DataSet {
   return !query ? dataset : fuse.search(query, { limit: 10 }).map((result) => result.item);
 }
 
+export const getFaviconUrl = (domain: string | undefined) => {
+  if (domain) {
+    return `https://www.google.com/s2/favicons?sz=64&domain=${encodeURI(domain)}`;
+  }
+};
+
```
Comment (thomaspaulmann) [inline extensions/dot-new/src/index.tsx:20]: Something I just did locally was adding fallback images to this function in the Safari extension :D 

Here is the code snippet:

```
export const getFaviconUrl = (domain: string | undefined) => {
  if (!domain) {
    return Icon.Globe;
  }

  return {
    source: `https://www.google.com/s2/favicons?sz=64&domain=${encodeURI(domain)}`,
    fallback: Icon.Globe,
  };
};
```

File: extensions/dot-new/scripts/generate-dataset.ts (hunk 1 above)
Comment (thomaspaulmann) [reply extensions/dot-new/scripts/generate-dataset.ts:22]: Oh this is really smart, nice!

Comment (natterstefan) [reply extensions/dot-new/scripts/generate-dataset.ts:22]: > Oh this is really smart, nice!

Thank you :)

File: extensions/dot-new/src/index.tsx (hunk 2 above)
Comment (natterstefan) [reply extensions/dot-new/src/index.tsx]: Oh, thanks for sharing that @thomaspaulmann. Here's the update: https://github.com/raycast/extensions/pull/1254/commits/edb1acd5d04ac231a0cbd5d95c0c834aed5595af.

**Comment by sxn** [comment]
🎉

**Pull request merged**